paasta_tools/vitesscluster_tools.py

```diff
@@ -1,31 +1,581 @@
     replicas: int
+    resources: Dict[str, Any]
+
+
+class CellConfigDict(TypedDict, total=False):
+    name: str
+    gateway: GatewayConfigDict
+
+
+class VitessDashboardConfigDict(TypedDict, total=False):
+    cells: List[str]
+    affinity: Dict[str, Any]
+    extraEnv: List[Dict[str, Any]]
+    extraFlags: Dict[str, str]
+    extraLabels: Dict[str, str]
+    replicas: int
+    resources: Dict[str, Any]
+
+
+class VtAdminConfigDict(TypedDict, total=False):
+    cells: List[str]
+    apiAddresses: List[str]
+    affinity: Dict[str, Any]
+    extraEnv: List[Dict[str, Any]]
+    extraFlags: Dict[str, str]
+    extraLabels: Dict[str, str]
+    replicas: int
+    readOnly: bool
+    apiResources: Dict[str, Any]
+    webResources: Dict[str, Any]
+
+
+class VtTabletDict(TypedDict, total=False):
+    extraFlags: Dict[str, str]
+    resources: Dict[str, Any]
+
+
+class TabletPoolDict(TypedDict, total=False):
+    cell: str
+    name: str
+    type: str
+    affinity: Dict[str, Any]
+    extraLabels: Dict[str, str]
+    extraEnv: List[Dict[str, Any]]
+    extraVolumeMounts: List[Dict[str, Any]]
+    extraVolumes: List[Dict[str, Any]]
+    replicas: int
+    vttablet: VtTabletDict
+    externalDatastore: Dict[str, Any]
+    dataVolumeClaimTemplate: Dict[str, Any]
+
+
+class ShardTemplateDict(TypedDict, total=False):
+    databaseInitScriptSecret: Dict[str, str]
+    tabletPools: List[TabletPoolDict]
 
 
-class VitessDeploymentConfig(LongRunningServiceConfig):
-    config_dict: VitessDeploymentConfigDict
+class PartitioningValueDict(TypedDict, total=False):
+    parts: int
+    shardTemplate: ShardTemplateDict
+
+
+class KeyspaceConfigDict(TypedDict, total=False):
+    durabilityPolicy: str
+    turndownPolicy: str
+    partitionings: List[Dict[str, PartitioningValueDict]]
+    name: str
+
+
+def get_updated_environment_variables(env_vars: Dict[str, Any]) -> List[Dict[str, Any]]:
+    """
+    Helper function to take in key value pairs of environment variables and return a list of dicts
+    """
+    updated_environment_variables: List[Dict[str, Any]] = []
+    for env_key, env_value in env_vars.items():
+        if isinstance(env_value, str):
+            updated_environment_variables.append({"name": env_key, "value": env_value})
+        elif isinstance(env_value, dict):
+            updated_environment_variables.append(
+                {"name": env_key, "valueFrom": env_value}
+            )
+        else:
+            log.error(f"Invalid environment variable {env_key}={env_value}")
+    return updated_environment_variables
+
+
+def get_cell_config(
+    cell: str,
+    region: str,
+    vtgate_resources: Dict[str, str],
+    env: List[Dict[str, Any]],
+    labels: Dict[str, str],
+    node_affinity: dict,
+) -> CellConfigDict:
+    """
+    get vtgate config
+    """
+    try:
+        replicas = int(vtgate_resources.get("replicas"))
+    except ValueError:
+        log.error("Invalid replicas value for vtgate")
+        replicas = 1
```

Comment:
it might be worth adding a schema at `paasta_tools/cli/schemas/` so that we can ensure that the right values are in place (and that they're in the correct format (i.e., ints) :))

Reply:
otherwise, 
```suggestion
    try:
        replicas = int(vtgate_resources.get("replicas"))
    except ValueError:
        log.error("Invalid replicas value for vtgate - defaulting to a single replica.")
        replicas = 1
```
will make it a little clearer to anyone looking at logs that this is something that was recoverable

(and we'd want to do this for the other error messages)

Reply:
that said, i'd still encourage adding a schema so that this isn't actually possible to trigger :)

Reply:
I'm willing to add schema to ensure we have specifications in the right type and avoid unexpected values to be passed. I was thinking of adding the whole json schema from kuberenetes-schema.json and add the relevant vitess properties and their schemas. Would that be a fine compromise?

Also, avoided hitting this mypy error which made me add int type conversion in the first place with https://github.com/Yelp/paasta/pull/3889/commits/2f6ecc76417760dbd1cb424c3b61dabd5c83ea09